src/transport_performance/utils/raster.py

```diff
@@ -32,9 +36,9 @@ def merge_raster_files(
 
     Parameters
     ----------
-    input_dir : str
+    input_dir : Union[str, pathlib.Path]
         Directory containing input raster files.
-    output_dir : str
+    output_dir : Union[str, pathlib.Path]
         Directory to write output, merged raster file.
     output_filename : str
```

Comment:
### Suggested Changes  
Type checking should be implemented for `output_filename`. This is because it is currently being used (in `os.path.join()`)  before being defended against in any way.  
Tests could also be included for this defence. This would fit well in one of the parameterised pytests (starting at line 237 in `tests/utils/test_raster.py`)

### Implementation (optional)  

```python
_type_defence(output_filename, "output_filename", str)
```

Reply:
Great catch! I'll implement this.

Reply:
This should now be included in commit a2917d2. There is a corresponding unit test added too.

Let me know if this is OK and we can resolve this conversation. Thanks again!

Reply:
Going to resolve this conversation and approve this PR. Great stuff